### ISSUE TITLE
hvplot of pandas dataframe doesn't follow pandas syntax for controlling markers

### ISSUE THREAD

**Issue by rjp23** [pyviz/hvplot#168]
In pandas "marker" or "style" control the plot symbols

e.g.
`df.loc['a'].plot(marker='o')`

but with hvplot:

`df.loc['a'].hvplot(marker='o')`

gives

> WARNING:root:main: marker option not found for line plot; similar options include: []

**Comment by jbednar** [comment]
It sounds like the Pandas .plot api is inferring that if you set a marker shape you want a points and not the default of a line plot?  Or does Pandas plot a line and _also_ add markers to it in this case?

**Comment by rjp23** [comment]
Yep, the latter. It'll plot points along the line in Pandas.

**Comment by jbednar** [comment]
Ok, I'll change the issue title then.

**Comment by rjp23** [comment]
Sorry? The title's correct isn't it?

**Comment by rjp23** [comment]
I guess it's not just adding the markers, the the whole styling of the plots (changing the colour, changing the marker type, changing the linestyle, using the style keyword, etc)

**Comment by philippjfr** [comment]
The difference is really in matplotlib since the default plot call plots lines but also supports markers. I do agree that we should emulate this behavior in hvplot though.

**Comment by jbednar** [comment]
Should hvplot be generating an overlay of line and a blank marker, by default, to support this behavior?

**Comment by philippjfr** [comment]
>Should hvplot be generating an overlay of line and a blank marker, by default, to support this behavior?

Seems inefficient and pretty pointless.

**Comment by jbednar** [comment]
Just wondering how to support it efficiently without adding a lot of complexity, then.

**Comment by DanielRudnicki** [comment]
I've been looking for this feature, I guess  that it's not been implemented yet (? correct me pls).
There is quite a big difference in a way how matplotlib treats plot() and scatter() on interactive plots.
The first one is much more memory-friendly. It would be great to have in in hvplot, however the hoovering would be useless then (since plot() result is single object, and not many points)

**Comment by philippjfr** [comment]
@holoviz/triaging-team This is a clear obstacle for new users so I think we should support this. Any other opinions out there?

**Comment by maximlt** [comment]
I agree hvPlot should support that. I assume it doesn't make sense to add support to that to HoloViews that provides more 'atomic' elements compared to hvPlot.

I suggest adding basic support to markers to `df.hvplot.line`, supporting `marker` (`'o'`, etc.), `markersize` and `markercolor`. Internally it would return an overlay of Curve and Scatter. For more more (e.g. not having a marker every point) I suggest documenting how to do that by creating the overlay explicitly.

**Comment by redlegjed** [comment]
Did anything ever happen about this?

I've been using hvplot for years now, mainly for simple line plots. One of the most frustrating things has been the inability to add markers as an optional argument. It's the only thing that matlab and matplotlib do better. I know you can put together a line and a scatter plot like this:

```python
df.hvplot.line(x="x", y="y") *\
df.hvplot.scatter(x="x", y="y", marker="*")
```

That's fine for one or two curves, but if you have 10 curves then it's really tedious and makes the code less readable.  It's even more annoying if you can generate the plots with a simple list comprehension like this:

```python
# Example using a xarray dataset
plots = [ ds.sel(TdegC=T).Reading.hvplot(by='setting') for T in temperatures]
hv.Layout(plots)
```

It's a lot more code to add markers to this because in order to keep the line and markers the same colour they have to be added in pairs like this:

```python
# Adding markers
plots = []
for T in temperatures:
    plots.append(ds.sel(TdegC=T).Reading.hvplot(by='setting')
    plots.append(ds.sel(TdegC=T).Reading.hvplot.scatter(by='setting')

hv.Layout(plots)
```

Typically I just use the basic .hvplot, so my ideal syntax would be:

```python
df.hvplot(x='x', y='y') # No markers
df.hvplot(x='x', y='y', marker='*') # With markers

# This would also be acceptable
df.hvplot.line(x='x', y='y', marker='*') 
```
